app/helpers/track_image.rb

```diff
@@ -2,15 +2,15 @@ module ExercismWeb
   module Helpers
     module TrackImage
       def track_icon(track_id, size=40)
-        '<img src="%s" width="%d" height="%d">' % [track_png(track_id), size, size]
+        '<img src="%s" width="%d" height="%d">' % [track_image(track_id), size, size]
       end
 
       private
 
-      def track_png(track_id)
-        img = "/img/tracks/%s.png" % track_id
-        if File.exist?("./public/%s" % img)
-          img
+      def track_image(track_id)
+        track = ::X::Track.find(track_id)
+        if track.image
+          track.image
         else
           "/img/e_red.png"
         end
```

Comment:
Oh, wow. That was a sloppy conditional on my part. Great catch, thank you!

Will change that depending on what gets decided in https://github.com/exercism/x-api/pull/112. We might serve the default image from x-api, which would simplify that method to just `track.image`.